### PR TITLE
fix mesh2d_manual example

### LINKED ISSUE #15847: `mesh2d_manual` example doesn't render correctly
## Bevy version

main, since #15524

## Relevant system information

SystemInfo { os: "MacOS 14.5 ", kernel: "23.5.0", cpu: "Apple M1 Max", core_count: "10", memory: "64.0 GiB" }
AdapterInfo { name: "Apple M1 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }

Also tested win11+4080

## What you did

`cargo run --example mesh2d_manual`

## What went wrong

Example should display a yellow star. But instead it displays some wispy blue lines.

### PR DESCRIPTION
**Pull request by awtterpip**
# Objective

- Fixes #15847 

## Solution

- Change mesh2d_manual to use the proper vertex format for color.

## Testing

- The mesh2d_manual example now works for me, tested on linux

### CODE REVIEW

File: examples/2d/mesh2d_manual.rs [hunk 1]
```diff
@@ -264,7 +264,7 @@ fn vertex(vertex: Vertex) -> VertexOutput {
     let model = mesh2d_functions::get_world_from_local(vertex.instance_index);
     out.clip_position = mesh2d_functions::mesh2d_position_local_to_clip(model, vec4<f32>(vertex.position, 1.0));
     // Unpack the `u32` from the vertex buffer into the `vec4<f32>` used by the fragment shader
```
Comment (rparrett) [inline examples/2d/mesh2d_manual.rs:266]: This comment is no longer needed

**Comment by rparrett** [comment]
This looks good, but I'm curious about why we changed the vertex attribute format in that migration. I had the impression that we were intentionally using a custom vertex attribute in that example.

File: examples/2d/mesh2d_manual.rs (hunk 1 above)
Comment (alice-i-cecile) [inline examples/2d/mesh2d_manual.rs:266]:
```suggestion
```

**Comment by rparrett** [comment]
Would like to know if the change away from the custom vertex attribute in #15847 was intentional or desired before approving.

This fixes the issue, but it could also be fixed by switching back to the custom `u32` based color, probably. And I think that example may have been doing that intentionally.

**Comment by Jondolf** [comment]
@rparrett It was half-intentional, but can be reverted. I originally changed it to fix a crash caused by conflicting vertex color formats with default mesh materials (see [this](https://github.com/bevyengine/bevy/pull/15524#discussion_r1783183321)) but then fixed it in a different way as per Cart's suggestion. I forgot to change it back and thought it was cleaner like that anyway, since it's the default format that [`Mesh::ATTRIBUTE_COLOR`](https://github.com/bevyengine/bevy/blob/813c75958df8a518353c6b50cb62a27353bbfd51/crates/bevy_mesh/src/mesh.rs#L176-L177) uses.

My bad though! Didn't anticipate that it would break things or be particularly undesirable (I recall testing it and it worked like normal...)

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
#15883 was selected over this PR because it better preserves the intent of the example.

**Pull request closed without merging**